Application/EdFi.Ods.Api/Helpers/AssemblyLoaderHelper.cs

```diff
@@ -146,6 +147,50 @@ public static IEnumerable<string> FindPluginAssemblies(string pluginFolder, bool
                 yield break;
             }
 
+            var apiModelFiles = Directory.GetFiles(pluginFolder, "ApiModel-EXTENSION.json", SearchOption.AllDirectories);
+
+            var physicalNames = new List<KeyValuePair<string, string>>();
+
+            Array.Sort(apiModelFiles, StringComparer.InvariantCulture);
+
+            foreach (var apiModelFilePath in apiModelFiles)
+            {
+                var myJsonString = File.ReadAllText(apiModelFilePath);
+
+                var myJsonObject = JObject.Parse(myJsonString);
+
+                JToken physicalNameJToken = myJsonObject.SelectToken("$.schemaDefinition.physicalName");
+
+                if (physicalNameJToken != null)
+                {
+                    string[] foldernames = apiModelFilePath.Split('\\');
+                    foreach (string folder in foldernames)
+                    {
+                        if (folder.Contains("Extensions"))
+                        {
+                            string key = physicalNameJToken.Value<string>();
+                            var element = new KeyValuePair<string, string>(key, folder);
+                            physicalNames.Add(element);
+                        }
+                    }
+                } 
+            }
+
+            string duplicatePhysicalName = "";
+            string duplicatePluginFolder = "";
+            foreach (var physicalName in physicalNames)
+            {
+                if(physicalName.Key.EqualsIgnoreCase(duplicatePhysicalName))
+                {
+                    _logger.Debug($"found duplicate extension schema name '{physicalName.Key}' in plugin folder." +
+                        $" You will be able to deploy only one of the following plugins '{physicalName.Value}' and '{duplicatePluginFolder}' folder name." +
+                        $" Please remove the conflicting plugins and retry");
+                    throw new Exception();
+                }
+                duplicatePhysicalName = physicalName.Key;
+                duplicatePluginFolder = physicalName.Value;
+            }

```

Comment:
I see a couple issues here:
- I would expect the log to be an `error` not a `debug` message
- the position of the `throw` stops execution preventing the additional messages from being logged defeating the purpose of having multiple log messages
- if there are three distinct conflicting schema names there should only be three messages
- the messages are not always correct. here is a sample of the output with three duplicate plugins. highlighting the incorrect ones:
	- found duplicate extension schema name 'homograph' in plugin folder. You will be able to deploy only one of the following plugins 'EdFi.Suite3.Ods.Extensions.Homograph.5.3.11' and 'EdFi.Suite3.Ods.Extensions.Homograph.5.3.11 copy' folder name. Please remove the conflicting plugins and retry.
	- found duplicate extension schema name `'sample'` in plugin folder. You will be able to deploy only one of the following plugins `'EdFi.Suite3.Ods.Extensions.Sample.5.3.10' and 'EdFi.Suite3.Ods.Extensions.Homograph.5.3.11'` folder name. Please remove the conflicting plugins and retry.
	- found duplicate extension schema name `'tpdm'` in plugin folder. You will be able to deploy only one of the following plugins `'EdFi.Suite3.Ods.Extensions.TPDM.1.0.0.5.3.11' and 'EdFi.Suite3.Ods.Extensions.Homograph.5.3.11'` folder name. Please remove the conflicting plugins and retry.
	- found duplicate extension schema name 'sample' in plugin folder. You will be able to deploy only one of the following plugins 'EdFi.Suite3.Ods.Extensions.Sample.5.3.10' and 'EdFi.Suite3.Ods.Extensions.Sample.5.3.10 copy' folder name. Please remove the conflicting plugins and retry.
	- found duplicate extension schema name `'tpdm'` in plugin folder. You will be able to deploy only one of the following plugins `'EdFi.Suite3.Ods.Extensions.TPDM.1.0.0.5.3.11' and 'EdFi.Suite3.Ods.Extensions.Sample.5.3.10'` folder name. Please remove the conflicting plugins and retry.
	- found duplicate extension schema name 'tpdm' in plugin folder. You will be able to deploy only one of the following plugins 'EdFi.Suite3.Ods.Extensions.TPDM.1.0.0.5.3.11 copy' and 'EdFi.Suite3.Ods.Extensions.TPDM.1.0.0.5.3.11' folder name. Please remove the conflicting plugins and retry.
	- found duplicate extension schema name 'tpdm' in plugin folder. You will be able to deploy only one of the following plugins 'EdFi.Suite3.Ods.Extensions.TPDM.1.0.0.5.3.11' and 'EdFi.Suite3.Ods.Extensions.TPDM.1.0.0.5.3.11 copy' folder name. Please remove the conflicting plugins and retry.

Reply:
2021-09-23 09:16:51,878 [1] ERROR EdFi.Ods.Api.Helpers.AssemblyLoaderHelper - MESSAGE: found duplicate extension schema name 'homograph' in plugin folder. You will be able to deploy only one of the following plugins 'EdFi.Suite3.Ods.Extensions.Homograph.5.3.11' and 'EdFi.Suite3.Ods.Extensions.Homograph.5.3.11 - Copy' folder name. Please remove the conflicting plugins and retry
 2021-09-23 09:22:36,303 [1] ERROR EdFi.Ods.Api.Helpers.AssemblyLoaderHelper - MESSAGE: found duplicate extension schema name 'sample' in plugin folder. You will be able to deploy only one of the following plugins 'EdFi.Suite3.Ods.Extensions.Sample.5.3.10' and 'EdFi.Suite3.Ods.Extensions.Sample.5.3.10 - Copy' folder name. Please remove the conflicting plugins and retry
 2021-09-23 09:22:38,813 [1] ERROR EdFi.Ods.Api.Helpers.AssemblyLoaderHelper - MESSAGE: found duplicate extension schema name 'tpdm' in plugin folder. You will be able to deploy only one of the following plugins 'EdFi.Suite3.Ods.Extensions.TPDM.1.0.0.5.3.11' and 'EdFi.Suite3.Ods.Extensions.TPDM.1.0.0.5.3.11 - Copy' folder name. Please remove the conflicting plugins and retry

Reply:
Above  is the log error message for given use case of 3 conflicts in schema name